File: php/features.php
<?php
declare(strict_types=1);

// Claude SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ClaudeFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ClaudeBaseFeature();
            case "test":
                return new ClaudeTestFeature();
            default:
                return new ClaudeBaseFeature();
        }
    }
}

File: py/features.py
# Claude SDK feature factory

from feature.base_feature import ClaudeBaseFeature
from feature.test_feature import ClaudeTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ClaudeBaseFeature(),
        "test": lambda: ClaudeTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
